rls/src/main/java/io/grpc/rls/RlsLoadBalancer.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2020 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.rls;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.MoreObjects;
+import io.grpc.ChannelLogger.ChannelLogLevel;
+import io.grpc.ConnectivityState;
+import io.grpc.LoadBalancer;
+import io.grpc.Status;
+import io.grpc.rls.internal.AdaptiveThrottler;
+import io.grpc.rls.internal.CachingRlsLbClient;
+import io.grpc.rls.internal.ChildLbResolvedAddressFactory;
+import io.grpc.rls.internal.LbPolicyConfiguration;
+import javax.annotation.Nullable;
+
+/**
+ * Implementation of {@link LoadBalancer} backed by route lookup service.
+ */
+final class RlsLoadBalancer extends LoadBalancer {
+
+  private final Helper helper;
+  @Nullable
+  private LbPolicyConfiguration lbPolicyConfiguration;
+  @Nullable
+  private CachingRlsLbClient routeLookupClient;
+
+  RlsLoadBalancer(Helper helper) {
+    this.helper = checkNotNull(helper, "helper");
+  }
+
+  @Override
+  public void handleResolvedAddresses(ResolvedAddresses resolvedAddresses) {
+    LbPolicyConfiguration lbPolicyConfiguration =
+        (LbPolicyConfiguration) resolvedAddresses.getLoadBalancingPolicyConfig();
+    if (lbPolicyConfiguration != null
```

Comment:
Is it possible that `lbPolicyConfiguration  == null`? If possible, is no-op the right behavior in that case?

Reply:
hmm i think it is possible if using no service config and default lb policy is rls. throwing NPE instead of ignoring it.